lib/alephant/logger/json.rb

```diff
@@ -41,6 +43,10 @@ def write(hash)
         @log_file.write(::JSON.generate(hash) + "\n")
       end
 
+      def writeable?(message_level)
+        Alephant::Logger::Level.new(message_level).logs?(@desired_write_level)
```

Comment:
~I wonder if we should pass in the `desired_write_level` on intialize, and then `message_level` into `logs?` or a `should_log?`. This would allow us to memoize the level object as the desired_write_level doesn't change between levels.~

I think Level, should be a LevelController, or a LevelManager as it defines logic of the desired level and the current message level, then does some validation to check if we should write or not.

```rb
@level_manager ||= Alephant::Logger::LevelSupervisor.new(@desired_write_level)
@level_manager.logs?(message_level)
```

Actually, the level class should be a static class, as it doesn't need to store any state. So we can always pass in both levels and always get the same output. We could use named params, to clarify the two levels, but it doesn't matter if you prefer to or not. We wouldn't need class variables either, as this would reduce the need to initialize separately and keep that variable for use when calling the logs method.

```rb
module Alephant
  module Logger
    class LevelSupervisor
      class << self
        def should_log?(message_level: message_level, desired_write_level: desired_write_level)
          message_level_index(message_level) >= desired_write_level_index(desired_write_level)
        end

        # other methods ...
      end
    end
  end
end

def writeable?(message_level)
  Alephant::Logger::LevelSupervisor.should_log?(message_level: message_level, desired_write_level: desired_write_level)
end
```

Reply:
@woodyblah, a good opportunity for naming something that manages/validates logging levels here. ^

Reply:
~~`MessageLevelWritablilityIndicator`~~ ~~`LoggingLevelComparator`~~ 

`WriteLevelChecker`? or `LoggingLevelChecker`? Or even just `LevelChecker` or `LevelComparator`?

Reply:
Good shouts both. 🙂  Not sure which name to go with. `LevelSupervisor` sounds good to me; on the other hand, looks like we should name it according to its responsibility i.e. `WriteLevelChecker`?